src/main/java/net/dries007/tfc/common/recipes/BloomeryRecipe.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * Licensed under the EUPL, Version 1.2.
+ * You may obtain a copy of the Licence at:
+ * https://joinup.ec.europa.eu/collection/eupl/eupl-text-eupl-12
+ */
+
+package net.dries007.tfc.common.recipes;
+
+import javax.annotation.Nullable;
+
+import com.google.gson.JsonObject;
+import net.minecraft.network.FriendlyByteBuf;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.crafting.RecipeSerializer;
+import net.minecraft.world.item.crafting.RecipeType;
+import net.minecraft.world.level.Level;
+
+import net.minecraftforge.fluids.FluidStack;
+
+import net.dries007.tfc.common.recipes.ingredients.FluidStackIngredient;
+import net.dries007.tfc.common.recipes.ingredients.ItemStackIngredient;
+import net.dries007.tfc.common.recipes.inventory.ItemStackInventory;
+import net.dries007.tfc.util.JsonHelpers;
+
+public class BloomeryRecipe implements ISimpleRecipe<ItemStackInventory>
+{
+    private final ResourceLocation id;
+    private final FluidStackIngredient fluidStackIngredient;
+    private final ItemStackIngredient itemStackIngredient;
```

Comment:
Why is this an `ItemStackIngredient` and not an `Ingredient`. It's being heated so by definition it needs to be stack size one.